### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by KareemWilson**
## **In this branch, @samuelmoiwa and I did the following:** 

**1. Install CanCanCan in our project.**
**2. Add a role column to the users table.**
**3. Implement the following functionalities**

- The post-deleting functionality
- The comment-deleting functionality

**4. Add the following Authorizations:**

- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin") 
- A user can delete a comment if it is theirs or if they have an admin role

### CODE REVIEW

File: app/views/users/_post.html.erb [hunk 1]
```diff
@@ -4,9 +4,13 @@
...
         <%= form_with url: user_post_likes_path(post_id: post.id, user_id: user.id ), method: :post do |f| %>
-        <%=f.hidden_field :post_id, value: post.id %>
+        <%= f.hidden_field :post_id, value: post.id %>
         <%= f.submit :Like %> 
-        <%end%>
+        <%end%><br/>
+            <%= form_with url: user_post_path(user, post), method: :delete do |f| %>
+            <%= f.hidden_field :post_id, value: post.id %>
+            <%= f.submit :Delete %> 
+            <%end%><br/>
```
Comment (tugsem) [inline app/views/users/_post.html.erb]: - [ ] Good job on applying the authorization properly! 💯   However, as you can read below, I shouldn't see the delete buttons if I am not authorized to do. Kindly add this functionality as well. 
image
image
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hi @KareemWilson  & @samuelmoiwa , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️Good description of project
✔️Linters
✔️Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/_post.html.erb (hunk 1 above)
Comment (KareemWilson) [reply app/views/users/_post.html.erb]: I did it, Thank you so much

**Review by Ismailco — APPROVED** [review]
### Hi @KareemWilson, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**